src/robotFactory.js

```diff
@@ -1,10 +1,88 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
```

Comment:
you should put each prop on the next line if there are more than 2. Also you can use destructuring in coords
```suggestion
  constructor(
    name,
    weight,
    { x = 0, y = 0 },
    chipVersion
  ) {
```